dissect/target/plugins/apps/webservers/apache.py

```diff
@@ -3,25 +3,121 @@
+"""
+
+COMMON_REGEX = r"""
+    \[(?P<ts>[^\]]*)\]                  # Timestamp including milliseconds.
+    \s
+    (\[(?P<pid>[0-9]+)\]\s)?            # The process ID of the child that serviced the request.
+    "
+    (?P<method>.*?)                     # The HTTP Method used for the request.
+    \s
+    (?P<uri>.*?)                        # The HTTP URI of the request.
+    \s
+    ?(?P<protocol>HTTP\/.*?)?           # The request protocol.
+    "
+    \s
+    (?P<status_code>\d{3})              # The HTTP Status Code of the response.
+    \s
+    (?P<bytes_sent>-|\d+)               # Bytes sent, including headers
+"""
+
+RESPONSE_TIME_REGEX = r"""
+(
+    "
+    Time:\s
+    (?P<response_time>.*?)              # Time taken to serve the response, including a unit of measurement.
+    "
+)
+"""  #
+
+COMMON_ERROR_REGEX = r"""
+    \[
+        (?P<ts>[^\]]*)                  # Timestamp including miliseconds.
+    \]
+    \s
+    \[
+        (?P<module>[^:]*)               # Name of the module logging the message.
+        \:
+        (?P<level>[^]]*)                # Loglevel of the message.
+    \]
+    \s
+    \[
+        pid\s(?P<pid>\d*)               # Process ID of current process
+        (\:tid\s(?P<tid>\d*))?          # Thread ID of current thread (optional).
+    \]
+    \s
+    ((?P<error_source>[^\:]*)\:\s)?     # Source file name and line number of the log call (optional).
+    (
+        \[
+            client\s(?P<client>[^]]+)   # Client IP address and port of the request (optional).
+        \]\s
+    )?
+    ((?P<error_code>\w+)\:\s)?          # APR/OS error status code and string (optional).
+    (?P<message>.*)                     # The actual log message.
+"""
 
 
 class LogFormat(enum.Enum):
-    VHOST_COMBINED = re.compile(rf"(?P<server_name>.*?):(?P<port>.*) {COMMON_REGEX} {REFERER_USER_AGENT_REGEX}")
-    COMBINED = re.compile(rf"{COMMON_REGEX} {REFERER_USER_AGENT_REGEX}")
-    COMMON = re.compile(COMMON_REGEX)
+    COMMON = re.compile(rf"{REMOTE_REGEX}\s{COMMON_REGEX}", re.VERBOSE)
+    VHOST_COMBINED = re.compile(
+        rf"(?P<server_name>.*?):(?P<port>.*)\s{REMOTE_REGEX}\s{COMMON_REGEX}\s{REFERER_USER_AGENT_REGEX}",
+        re.VERBOSE,
+    )
+    COMBINED = re.compile(rf"{REMOTE_REGEX}\s{COMMON_REGEX}\s{REFERER_USER_AGENT_REGEX}", re.VERBOSE)
+
+    CITRIX_NETSCALER_COMBINED_RESPONSE_TIME = re.compile(
+        rf"{REMOTE_REGEX}\s{COMMON_REGEX}\s{REFERER_USER_AGENT_REGEX}\s{RESPONSE_TIME_REGEX}",
+        re.VERBOSE,
+    )
+    CITRIX_NETSCALER_COMBINED_RESPONSE_TIME_WITH_HEADERS = re.compile(
+        rf"""
+        (?P<remote_ip>.*?)          # Client IP address of the request.
+        \s
+        ->
+        \s
+        (?P<local_ip>.*?)           # Local IP of the Netscaler.
+        \s
+        (?P<remote_logname>.*?)     # Remote logname (from identd, if supplied).
+        \s
+        (?P<remote_user>.*?)        # Remote user if the request was authenticated.
```

Comment:
Maybe you can think of an elegant way to re use the `REMOTE_REGEX` pattern here, since it is quite similar. So use a function that returns a specific version based on a boolean or something in that direction.

Reply:
I agree it's quite similar, but didn't see a way of re-using components here. Considering the regex pattern has been moved to a separate file, I think the code overlap is a little less glaring. Still open to suggestions on this though.